modules/host/host.go

```diff
@@ -31,9 +31,8 @@ type contractObligation struct {
 	LastRevisionTxn types.Transaction
 	Path            string // Where on disk the file is stored.
 
-	// each obligation needs a mutex to prevent simultaneous revisions to the
-	// same obligation
-	mu *sync.Mutex
+	// revisions must happen in serial
```

Comment:
hmm. Then how do we parallelize it? Seems like it's actually a really difficult problem, I'm guessing you were running into problems earlier because you weren't using the mutex correctly. There were undectable dataraces.

'thread 1 reads host:rev1'
'thread 2 reads host:rev1'
[work happens]
'thread2 grabs lock, writes host:rev1:rev2a'
'thread1 grabs lock, writes host:rev1:rev2b'
---> problem, despite no actual data races. If you want to parallelize revisions, you'll need to be fancier about it.